packages/backend-function/src/factory.ts

```diff
@@ -31,17 +31,25 @@ import { FunctionEnvironmentTypeGenerator } from './function_env_type_generator.
 import { AttributionMetadataStorage } from '@aws-amplify/backend-output-storage';
 import { fileURLToPath } from 'node:url';
 import { AmplifyUserError, TagName } from '@aws-amplify/platform-core';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
 
 const functionStackType = 'function-Lambda';
 
 /**
  * Entry point for defining a function in the Amplify ecosystem
  */
 export const defineFunction = (
-  props: FunctionProps = {}
+  props?: FunctionProps | ((scope: Construct) => lambda.Function)
 ): ConstructFactory<
   ResourceProvider<FunctionResources> & ResourceAccessAcceptorFactory
-> => new FunctionFactory(props, new Error().stack);
+> => {
+  if (props === undefined || props === null) {
```

Comment:
we should continue to pass props as-is into the FunctionFactory and move this if statement inside that class